components/tel/Core/Src/imu.h

```diff
@@ -0,0 +1,95 @@
+#include "i2c.h"
+#include "usart.h"
+
+enum GyroType {
+	GYRO_X,
+	GYRO_Y,
+    GYRO_Z
+};
+
+enum AccelType {
+	ACCEL_X,
+	ACCEL_Y,
+	ACCEL_Z
+};
+
+typedef struct {
+  uint8_t imu_type;
+  uint8_t dimension;
+  uint8_t data[4];
+} IMU_msg_t;
+
+void initIMU(void);
+float gyro(enum GyroType type);
+float accel(enum AccelType type);
+void transmitData(uint8_t* data, uint16_t size);
+
+#define DEVICE_ADDRESS ((0x6A)<<1)
+
+
+#define UART_TIMEOUT 1000
+
+void transmitData(uint8_t* data, uint16_t size)
+{
+	HAL_UART_Transmit(&huart3, data, size, UART_TIMEOUT);
+	HAL_Delay(1000);
+}
```

Comment:
Whether you move these functions to be in a declaration + definition in a c/h file or not, make sure to add some function documentation. I don't remember what this is and it isn't called anywhere in the project. Remove?